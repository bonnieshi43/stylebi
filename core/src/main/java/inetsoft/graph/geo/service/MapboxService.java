/*
 * inetsoft-core - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
package inetsoft.graph.geo.service;

import inetsoft.sree.SreeEnv;

import javax.json.*;
import javax.net.ssl.HttpsURLConnection;
import java.awt.*;
import java.io.IOException;
import java.net.URL;
import java.net.URLConnection;
import java.util.ArrayList;
import java.util.List;

public class MapboxService extends WebMapService {
   public MapboxService() {
      user = SreeEnv.getProperty("mapbox.user");
      token = SreeEnv.getProperty("mapbox.token");
      style = SreeEnv.getProperty("mapbox.style");
   }

   public MapboxService(boolean orgScoped) {
      user = SreeEnv.getProperty("mapbox.user", false, orgScoped);
      token = SreeEnv.getProperty("mapbox.token", false, orgScoped);
      style = SreeEnv.getProperty("mapbox.style", false, orgScoped);
   }

   public MapboxService(String user, String token, String style) {
      this.user = user;
      this.token = token;
      this.style = style;
   }

   /**
    * Load a static image using bbox.
    */
   @Override
   public Image loadImage(double lonMin, double lonMax, double latMin, double latMax,
                          int imgw, int imgh)
         throws IOException
   {
      // mapbox static image api:
      // https://api.mapbox.com
      // /styles/v1/{username}/{style_id}/static/{bbox}/{width}x{height}?access_token=...
      String urlstr = String.format("https://api.mapbox.com" +
                                    "/styles/v1/%s/%s/static/[%f,%f,%f,%f]/%dx%d?access_token=%s",
                                 user, style, lonMin, latMin, lonMax, latMax, imgw, imgh, token);
      return WebMapService.loadFromURL(urlstr);
   }

   /**
    * Get a list of all map styles for the user.
    */
   public List<MapboxStyle> getStyles(boolean withDefault) throws IOException {
      List<MapboxStyle> styles = new ArrayList<>();
      getStyles("https://api.mapbox.com/styles/v1/" + user + "?access_token=" + token, styles);

      if(withDefault) {
         styles.add(0, MapboxStyle.DEFAULT);
      }

      return styles;
   }

   private void getStyles(String urlstr, List<MapboxStyle> styles) throws IOException {
      URL url = new URL(urlstr);
      URLConnection conn = url.openConnection();

      try {
         String next = conn.getHeaderField("next");
         JsonReader parser = Json.createReader(conn.getInputStream());
         JsonArray arr = (JsonArray) parser.read();

         arr.stream().map(v -> new MapboxStyle(v)).forEach(v -> styles.add(v));

         if(next != null) {
            getStyles(next, styles);
         }
      }
      catch(IOException ex) {
         if(conn instanceof HttpsURLConnection) {
            if(((HttpsURLConnection) conn).getResponseCode() == 403) {
               throw new RuntimeException("Access token requires permission to STYLES:LIST secret scope");
            }
         }

         throw ex;
      }
   }

   /**
    * Get the tile size. One tile is required to show world at lowest zoom level.
    */
   @Override
   public int getTileSize() {
      return 512;
   }

   @Override
   public int getMaxSize() {
      return 1280;
   }

   @Override
   public boolean isDiscreteZoom() {
      return false;
   }

   private String user;
   private String token;
   private String style;
}
