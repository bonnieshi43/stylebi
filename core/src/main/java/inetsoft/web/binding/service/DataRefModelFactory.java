/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
package inetsoft.web.binding.service;

import inetsoft.uql.erm.DataRef;
import inetsoft.web.binding.drm.DataRefModel;

public abstract class DataRefModelFactory<R extends DataRef, M extends DataRefModel> {
   /**
    * Gets the data ref class supported by this factory.
    *
    * @return the data ref class.
    */
   public abstract Class<R> getDataRefClass();

   /**
    * Creates a new model instance for the specified dataRef.
    *
    * @param dataRef the dataRef interface.
    * @return a new model.
    */
   public abstract M createDataRefModel(R dataRef);
}
