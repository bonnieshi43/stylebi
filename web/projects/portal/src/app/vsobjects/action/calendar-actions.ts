/*
 * inetsoft-web - StyleBI is a business intelligence web application.
 * Copyright Â© 2024 InetSoft Technology (info@inetsoft.com)
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affrero General Public License
 * along with this program. If not, see <http://www.gnu.org/licenses/>.
 */
import { ContextProvider } from "../context-provider.service";
import { VSCalendarModel } from "../model/calendar/vs-calendar-model";
import { AbstractVSActions } from "./abstract-vs-actions";
import { ActionStateProvider } from "./action-state-provider";
import { AssemblyActionGroup } from "../../common/action/assembly-action-group";
import { DataTipService } from "../objects/data-tip/data-tip.service";
import { PopComponentService } from "../objects/data-tip/pop-component.service";
import { MiniToolbarService } from "../objects/mini-toolbar/mini-toolbar.service";

export class CalendarActions extends AbstractVSActions<VSCalendarModel> {
   constructor(model: VSCalendarModel, contextProvider: ContextProvider,
               securityEnabled: boolean = false,
               stateProvider: ActionStateProvider = null,
               dataTipService: DataTipService = null,
               popService: PopComponentService,
               miniToolbarService: MiniToolbarService = null)
   {
      super(model, contextProvider, securityEnabled, stateProvider,
            dataTipService, popService, miniToolbarService);
   }

   protected createMenuActions(groups: AssemblyActionGroup[]): AssemblyActionGroup[] {
      groups.push(new AssemblyActionGroup([
         {
            id: () => "calendar properties",
            label: () => "_#(js:Properties)...",
            icon: () => "fa fa-sliders",
            enabled: () => true,
            visible: () => this.composer
         },
         {
            id: () => "calendar show-format-pane",
            label: () => "_#(js:Format)...",
            icon: () => "fa fa-format",
            enabled: () => true,
            visible: () => this.composer
         },
      ]));
      groups.push(this.createDefaultEditMenuActions());
      groups.push(this.createDefaultOrderMenuActions());
      return super.createMenuActions(groups);
   }

   protected createToolbarActions(groups: AssemblyActionGroup[]): AssemblyActionGroup[] {
      groups.push(new AssemblyActionGroup([
         {
            id: () => "calendar toggle-year",
            label: () => this.model.yearView ? "_#(js:Switch To Month View)"
               : "_#(js:Switch To Year View)",
            icon: () => this.model.yearView ? "calendar-plus-icon" : "calendar-minus-icon",
            enabled: () => true,
            visible: () => (!this.model.dropdownCalendar || this.model.calendarsShown)
              && (this.isActionVisibleInViewer("Switch To Month View") &&
                  this.isActionVisibleInViewer("Switch To Year View"))
         },
         {
            id: () => "calendar toggle-double-calendar",
            label: () => this.model.doubleCalendar ? "_#(js:Switch To Simple View)"
               : (this.model.comparisonVisible || this.model.period) ? "_#(js:Switch To Range)"
               : "_#(js:Switch To Range)",
            icon: () => this.model.doubleCalendar ? "calendar-icon" : "calendar-double-icon",
            enabled: () => true,
            visible: () => (!this.model.dropdownCalendar || this.model.calendarsShown)
               && (this.isActionVisibleInViewer("Switch To Simple View") &&
                   this.isActionVisibleInViewer("Switch To Range") &&
                   this.isActionVisibleInViewer("Switch To Single Calendar") &&
                   this.isActionVisibleInViewer("Switch To Double Calendar"))
         },
         {
            id: () => "calendar clear",
            label: () => "_#(js:Clear Calendar)",
            icon: () => "eraser-icon",
            enabled: () => true,
            visible: () => this.isActionVisibleInViewer("Clear Calendar")
         },
         {
            id: () => "calendar toggle-range-comparison",
            label: () => this.model.period ? "_#(js:Switch To Date Range Mode)"
               : "_#(js:Switch To Comparison Mode)",
            // use different appropriate icon for switch to date range mode
            icon: () => this.model.period ? "date-range-icon" : "compare-icon",
            enabled: () => (this.model.comparisonVisible || this.model.period),
            visible: () => this.model.doubleCalendar
                           && (!this.model.dropdownCalendar || this.model.calendarsShown) &&
               (this.isActionVisibleInViewer("Switch To Date Range Mode") &&
                   this.isActionVisibleInViewer("Switch To Comparison Mode")) &&
               (this.model.comparisonVisible || this.model.period)
         },
         {
            id: () => "calendar multi-select",
            label: () => this.model.multiSelect ? "_#(js:Change to Single-select)"
               : "_#(js:Change to Multi-select)",
            icon: () => this.model.multiSelect ? "select-multi-icon" : "select-single-icon",
            enabled: () => true,
            visible: () => this.mobileDevice && !this.model.singleSelection &&
               this.isActionVisibleInViewer("Change to Single-select") &&
               this.isActionVisibleInViewer("Change to Multi-select")
         },
         {
            id: () => "calendar apply",
            label: () => "_#(js:Apply)",
            icon: () => "submit-icon",
            enabled: () => true,
            visible: () => (this.model.doubleCalendar || !this.model.submitOnChange)
                           && this.isActionVisibleInViewer("Apply")
         },
      ]));
      return super.createToolbarActions(groups, true);
   }

   protected getEditScriptActionId(): string {
      return "calendar edit-script";
   }
}
