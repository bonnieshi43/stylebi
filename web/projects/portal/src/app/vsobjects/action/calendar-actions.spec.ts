/*
 * This file is part of StyleBI.
 * Copyright (C) 2024  InetSoft Technology
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU Affero General Public License
 * along with this program.  If not, see <https://www.gnu.org/licenses/>.
 */
import { TestUtils } from "../../common/test/test-utils";
import { ComposerContextProviderFactory, ViewerContextProviderFactory } from "../context-provider.service";
import { CalendarActions } from "./calendar-actions";

describe("CalendarActions", () => {
   const createModel = () => TestUtils.createMockVSCalendarModel("Calendar1");

   const popService: any = { getPopComponent: jest.fn() };
   popService.getPopComponent.mockImplementation(() => "");

   it("check status of menu actions and toolbar actions of calendar in composer", () => {
      const expectedMenu = [
         [
            { id: "calendar properties", visible: true },
            { id: "calendar show-format-pane", visible: true }
         ],
         [
            { id: "vs-object copy", visible: true },
            { id: "vs-object cut", visible: true },
            { id: "vs-object remove", visible: true },
            { id: "vs-object group", visible: true },
            { id: "vs-object ungroup", visible: true }
         ],
         [
            { id: "vs-object bring-forward", visible: true },
            { id: "vs-object bring-to-front", visible: true },
            { id: "vs-object send-backward", visible: true },
            { id: "vs-object send-to-back", visible: true }
         ]
      ];

      const expectedToolbar = [
         [
            { id: "calendar toggle-year", visible: true },
            { id: "calendar toggle-double-calendar", visible: true },
            { id: "calendar clear", visible: true },
            { id: "calendar toggle-range-comparison", visible: false },
            { id: "calendar multi-select", visible: false },
            { id: "calendar apply", visible: false }
         ]
      ];

      const model = createModel();
      const actions = new CalendarActions(model, ComposerContextProviderFactory(), false, null, null, popService);
      const menuActions = actions.menuActions;
      const toolbarActions = actions.toolbarActions;

      expect(menuActions).toMatchSnapshot();
      expect(toolbarActions).toMatchSnapshot();
      expect(TestUtils.toString(toolbarActions[1].actions[0].label())).toBe("Switch To Year View");
      expect(TestUtils.toString(toolbarActions[1].actions[1].label())).toBe("Switch To Range");
   });

   it("check status of menu actions and toolbar actions of calendar in viewer and preview", () => {
      const expectedMenu = [
         [
            { id: "calendar properties", visible: false },
            { id: "calendar show-format-pane", visible: false }
         ],
         [
            { id: "vs-object copy", visible: false },
            { id: "vs-object cut", visible: false },
            { id: "vs-object remove", visible: false },
            { id: "vs-object group", visible: false },
            { id: "vs-object ungroup", visible: false }
         ],
         [
            { id: "vs-object bring-forward", visible: false },
            { id: "vs-object bring-to-front", visible: false },
            { id: "vs-object send-backward", visible: false },
            { id: "vs-object send-to-back", visible: false }
         ]
      ];

      const expectedToolbar = [
         [
            { id: "calendar toggle-year", visible: true },
            { id: "calendar toggle-double-calendar", visible: true },
            { id: "calendar clear", visible: true },
            { id: "calendar toggle-range-comparison", visible: false },
            { id: "calendar multi-select", visible: false },
            { id: "calendar apply", visible: false }
         ]
      ];

      //check status in viewer
      const model = createModel();
      const actions1 = new CalendarActions(model, ViewerContextProviderFactory(false), false, null, null, popService);
      const menuActions1 = actions1.menuActions;
      const toolbarActions1 = actions1.toolbarActions;

      expect(menuActions1).toMatchSnapshot();
      expect(toolbarActions1).toMatchSnapshot();

      //check status in preview
      const model2 = createModel();
      const actions2 = new CalendarActions(model2, ViewerContextProviderFactory(true), false, null, null, popService);
      const menuActions2 = actions2.menuActions;
      const toolbarActions2 = actions2.toolbarActions;

      expect(menuActions2).toMatchSnapshot();
      expect(toolbarActions2).toMatchSnapshot();
   });

   it("check status of toolbar actions of calendar in composer when is dropdown or double calendar", () => {
      //check toolbar when is dropdown
      const expectedToolbar1 = [
         [
            { id: "calendar toggle-year", visible: false },
            { id: "calendar toggle-double-calendar", visible: false },
            { id: "calendar clear", visible: true },
            { id: "calendar toggle-range-comparison", visible: false },
            { id: "calendar multi-select", visible: false },
            { id: "calendar apply", visible: false }
         ]
      ];

      const model = createModel();
      const actions = new CalendarActions(model, ComposerContextProviderFactory(), false, null, null, popService);
      const toolbarActions = actions.toolbarActions;
      model.dropdownCalendar = true;
      model.calendarsShown = false;

      expect(toolbarActions).toMatchSnapshot();

      //check toolbar when is double year calendar
      const expectedToolbar2 = [
         [
            { id: "calendar toggle-year", visible: true },
            { id: "calendar toggle-double-calendar", visible: true },
            { id: "calendar clear", visible: true },
            { id: "calendar toggle-range-comparison", visible: true },
            { id: "calendar multi-select", visible: false },
            { id: "calendar apply", visible: true }
         ]
      ];
      model.dropdownCalendar = false;
      model.calendarsShown = true;
      model.yearView = true;
      model.doubleCalendar = true;

      expect(toolbarActions).toMatchSnapshot();
      expect(TestUtils.toString(toolbarActions[1].actions[0].label())).toBe("Switch To Month View");
      expect(TestUtils.toString(toolbarActions[1].actions[1].label())).toBe("Switch To Simple View");
      expect(TestUtils.toString(toolbarActions[1].actions[3].label())).toBe("Switch To Comparison Mode");
   });

   //Bug #19986 should not display menu action when as data tip component
   it("should not display menu action when as data tip component", () => {
      const dataTipService: any = { isDataTip: jest.fn() };
      dataTipService.isDataTip.mockImplementation(() => true);
      const model = createModel();
      const actions = new CalendarActions(model, ViewerContextProviderFactory(false), false, null, dataTipService, popService);
      const menuActions = actions.menuActions;

      expect(menuActions.length).toBe(0);
   });
});