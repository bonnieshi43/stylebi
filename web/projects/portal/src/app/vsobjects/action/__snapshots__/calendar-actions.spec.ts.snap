// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in composer 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in composer 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in viewer and preview 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in viewer and preview 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in viewer and preview 3`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of menu actions and toolbar actions of calendar in viewer and preview 4`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of toolbar actions of calendar in composer when is dropdown or double calendar 1`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;

exports[`CalendarActions check status of toolbar actions of calendar in composer when is dropdown or double calendar 2`] = `
Array [
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
      Object {
        "action": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
  AssemblyActionGroup {
    "actions": Array [
      Object {
        "action": [Function],
        "childAction": [Function],
        "enabled": [Function],
        "icon": [Function],
        "id": [Function],
        "label": [Function],
        "visible": [Function],
      },
    ],
    "icon": [Function],
    "label": [Function],
  },
]
`;
